src/types/JSONSchema.ts

```diff
@@ -71,6 +72,8 @@ export interface LinkedJSONSchema extends JSONSchema {
 }
 
 export interface NormalizedJSONSchema extends LinkedJSONSchema {
+  $intersection?: NormalizedJSONSchema
+  $types: ReturnType<typeof typesOfSchema>
```

Comment:
Nit: please type this explicitly to keep it simple.
```suggestion
  $types: readonly [SchemaType, ...SchemaType[]]
```